cmd/libs/go2idl/go-to-protobuf/main.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// go-to-protobuf generates a Protobuf IDL from a Go struct, respecting any
+// existing IDL tags on the Go struct.
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
+	"k8s.io/kubernetes/cmd/libs/go2idl/namer"
+	"k8s.io/kubernetes/cmd/libs/go2idl/parser"
+
+	"github.com/golang/glog"
+	flag "github.com/spf13/pflag"
+)
+
+var (
+	inputPackages = flag.StringP("input-packages", "i" /*"k8s.io/kubernetes/pkg/api,k8s.io/kubernetes/pkg/apis/extensions"*/, "", "comma-separated list of directories to get input types from.")
+	packages      = flag.StringP("packages", "p", "k8s.io/kubernetes/pkg/api/unversioned=k8s.api.unversioned,k8s.io/kubernetes/pkg/api/v1=k8s.api.v1,k8s.io/kubernetes/pkg/apis/extensions/v1beta1=k8s.api.extensions.v1beta1", "comma-separated list of directories to get input types from.")
+	outputBase    = flag.StringP("output-base", "o", filepath.Join(os.Getenv("GOPATH"), "src"), "Output base; defaults to $GOPATH/src/")
```

Comment:
As @wojtek-t said elsewhere, we should really extract these out to a common place. I'll try to add that.